commit 28b9eb152750dfbb06d2e87ff9e929129aac4899
Author: srikanth716 <srikanthsri7447@gmail.com>
Date:   Mon Jun 23 14:28:15 2025 +0530

    [INJIVER-1100] Fix: Display full VC list after successful missing credential share
    
    Signed-off-by: srikanth716 <srikanthsri7447@gmail.com>

diff --git a/verify-ui/src/components/Home/VerificationSection/Result/VpSubmissionResult.tsx b/verify-ui/src/components/Home/VerificationSection/Result/VpSubmissionResult.tsx
index 2d32658c..a25016c3 100644
--- a/verify-ui/src/components/Home/VerificationSection/Result/VpSubmissionResult.tsx
+++ b/verify-ui/src/components/Home/VerificationSection/Result/VpSubmissionResult.tsx
@@ -30,6 +30,7 @@ const VpSubmissionResult: React.FC<VpSubmissionResultProps> = ({
   const vcStatus = verifiedVcs.length > 0 ? verifiedVcs[0].vcStatus : "INVALID";
   const selectedClaims: claim[] = useVerifyFlowSelector((state) => state.selectedClaims) || [];
   const isPartiallyShared = useVerifyFlowSelector((state) => state.isPartiallyShared );
+  const showResult = useVerifyFlowSelector((state) => state.isShowResult );
 
   const renderRequestCredentialsButton = (propClasses = "") => (
     <div className={`flex flex-col items-center lg:hidden ${propClasses}`}>
@@ -79,7 +80,7 @@ const VpSubmissionResult: React.FC<VpSubmissionResultProps> = ({
       )}
       <div className="relative">
         <div className="flex flex-col items-center space-y-4 lg:space-y-6 mt-[-60px] lg:mt-[-70px]">
-          {verifiedVcs.length > 0 && verifiedVcs.map(({ vc, vcStatus }, index) => (
+          {showResult && verifiedVcs.map(({ vc, vcStatus }, index) => (
             <DisplayVcCardView
               key={index}
               vc={vc}
diff --git a/verify-ui/src/redux/features/verify/vpVerificationState.ts b/verify-ui/src/redux/features/verify/vpVerificationState.ts
index f07908e4..f1770e64 100644
--- a/verify-ui/src/redux/features/verify/vpVerificationState.ts
+++ b/verify-ui/src/redux/features/verify/vpVerificationState.ts
@@ -14,10 +14,12 @@ const PreloadedState: VerifyState = {
   SelectionPanel: false,
   verificationSubmissionResult: [],
   selectedClaims: verifiableClaims?.filter((claim) => claim.essential),
+  originalSelectedClaims: verifiableClaims?.filter((claim) => claim.essential),
   unVerifiedClaims: [],
   sharingType: VCShareType.SINGLE,
   isPartiallyShared: false,
-  presentationDefinition:{
+  isShowResult: false,
+  presentationDefinition: {
     id: "c4822b58-7fb4-454e-b827-f8758fe27f9a",
     purpose:
       "Relying party is requesting your digital ID for the purpose of Self-Authentication",
@@ -57,14 +59,26 @@ const vpVerificationState = createSlice({
       const inputDescriptors = state.selectedClaims.flatMap((claim) => claim.definition.input_descriptors);
       state.presentationDefinition.input_descriptors = [...inputDescriptors];
       state.SelectionPanel = false;
+      state.isShowResult = false;
       state.activeScreen = VerificationSteps[state.method].ScanQrCode;
       state.verificationSubmissionResult = [];
       state.unVerifiedClaims = [];
     },
     verificationSubmissionComplete: (state, action) => {
-      const verifiedVCs = calculateVerifiedClaims(state.selectedClaims, action.payload.verificationResult);
-      state.verificationSubmissionResult.push(...verifiedVCs);
-      state.unVerifiedClaims = calculateUnverifiedClaims(state.selectedClaims, state.verificationSubmissionResult);
+      const newlyVerified = calculateVerifiedClaims(state.selectedClaims, action.payload.verificationResult);
+
+      const uniqueResult = [
+        ...state.verificationSubmissionResult,
+        ...newlyVerified.filter(
+          (vc) =>
+            !state.verificationSubmissionResult.some(
+              (existing) => existing.vc.type[1] === vc.vc.type[1]
+            )
+        ),
+      ];
+      state.verificationSubmissionResult = uniqueResult;
+      state.isShowResult = true;
+      state.unVerifiedClaims = calculateUnverifiedClaims(state.originalSelectedClaims, state.verificationSubmissionResult);
       state.isPartiallyShared = state.unVerifiedClaims.length > 0 && state.sharingType === VCShareType.MULTIPLE;
       state.activeScreen = state.isPartiallyShared
         ? VerificationSteps[state.method].RequestMissingCredential
@@ -88,6 +102,7 @@ const vpVerificationState = createSlice({
       state.status = "ACTIVE";
       state.sharingType = VCShareType.SINGLE;
       state.isPartiallyShared = false;
+      state.isShowResult = false;
       state.presentationDefinition.input_descriptors = [];
     },
   },
diff --git a/verify-ui/src/types/data-types.ts b/verify-ui/src/types/data-types.ts
index 62b165f3..a8ffa75d 100644
--- a/verify-ui/src/types/data-types.ts
+++ b/verify-ui/src/types/data-types.ts
@@ -155,7 +155,9 @@ export type VerifyState = {
   activeScreen: number;
   verificationSubmissionResult: VpSubmissionResultInt[];
   SelectionPanel: boolean;
+  isShowResult: boolean;
   selectedClaims: claim[];
+  originalSelectedClaims: claim[];
   unVerifiedClaims: claim[];
   sharingType: VCShareType;
   isPartiallyShared: boolean;
